### PR TITLE
Support both old and new diffusers import path

### PR DESCRIPTION
**Pull request by osanseviero**

### CODE REVIEW

File: trl/models/modeling_sd_base.py [hunk 1]
```diff
@@ -24,7 +24,10 @@
 from diffusers.loaders import AttnProcsLayers
 from diffusers.models.attention_processor import LoRAAttnProcessor
 from diffusers.pipelines.stable_diffusion.pipeline_stable_diffusion import rescale_noise_cfg
-from diffusers.utils.torch_utils import randn_tensor
+try:
+    from diffusers.utils import randn_tensor
+except ImportError:
+    from diffusers.utils.torch_utils import randn_tensor
```
Comment (younesbelkada) [inline trl/models/modeling_sd_base.py:30]:
```suggestion
import diffusers
_diffusers_greater_21_4 = hasattr(diffusers.utils, "randn_tensor")
if _diffusers_greater_21_4:
    from diffusers.utils import randn_tensor
else:
    from diffusers.utils.torch_utils import randn_tensor
```

**Review by younesbelkada — COMMENTED** [review]
Thanks ! I left a single comment, WDYT?

**Review by younesbelkada — COMMENTED** [review]
Thanks! can you run the styling checks?
```bash
make precommit
```

**Comment by younesbelkada** [comment]
Note the lazy import has been added in 0.21.0 and the latest version before that is 0.20.2

https://github.com/huggingface/diffusers/releases/tag/v0.21.0

File: trl/models/modeling_sd_base.py [hunk 2]
```diff
@@ -24,7 +24,14 @@
...
 from diffusers.pipelines.stable_diffusion.pipeline_stable_diffusion import rescale_noise_cfg
-from diffusers.utils.torch_utils import randn_tensor
+
+from ..import_utils import is_diffusers_greater_than
+
+
+if is_diffusers_greater_than("0.20.2"):
+    from diffusers.utils.torch_utils import randn_tensor
+else:
+    from diffusers.utils import randn_tensor
```
Comment (lvwerra) [inline trl/models/modeling_sd_base.py:34]: If I understood Patrick correctly, `randn_tensor` is an internal method that could be deprecated, delete, renamed at any time. Should we not use it and instead add our own?

Comment (younesbelkada) [reply trl/models/modeling_sd_base.py]: Makes sense yeah !

File: trl/torch_utils.py
Comment (lvwerra) [inline trl/torch_utils.py:1]: Why is there an empty file? :)

**Review by lvwerra — APPROVED** [review]
LGTM 🚀 Just that question about the empty file.

File: trl/torch_utils.py
Comment (younesbelkada) [reply trl/torch_utils.py:1]: oops will remove it

**Pull request merged**